includes/MediaBynderStreamWrapper.inc

```diff
@@ -9,17 +9,17 @@
 class MediaBynderStreamWrapper extends MediaReadOnlyStreamWrapper{
 
     public function interpolateUrl() {
-        $this->base_url = variable_get('media_bynder_url', '');
-
-        if ($parameters = $this->get_parameters()) {
-            return media_bynder_embed_url($parameters['i']);
-        } else {
-            $parameters = media_bynder_uri_parse_style_url($this->uri);
-            if($parameters){
-                return media_bynder_embed_url($parameters['i'], $parameters['style'], $parameters['effects']);
+        $parts = explode('/', $this->uri);
+      
+        foreach ($parts as $part => $part_key) {
+            if ($part_key == 'i'){
+                //idHash of URI
+                if (count($parts) >= ($part + 1)) {
```

Comment:
This looks a bit finnicky, maybe there's a more elegant way of dealing with the parts? To be honest I don't even know what "i" means in part_key and that's why I'm lost